### ISSUE TITLE
Prepare for WebExtension v3 Changes

### ISSUE THREAD

**Issue by danfinlay** [MetaMask/metamask-extension#6059]
Google is planning to make some changes to the WebExtension protocol, and the many changes are outlined in this document:

https://docs.google.com/document/d/1nPu6Wy4LWR66EFLeYInl3NzzhHzc-qnk4w4PX-0XMw8/edit#

Chromium issue: https://bugs.chromium.org/p/chromium/issues/detail?id=896897&desc=2

We will need to make these changes to continue to be a working WebExtension.

When the migration details are ready to be public, they will be available here:
https://docs.google.com/document/d/1FZthsZHCxa-93QCmyhU08Ptlh3EGqbjuUlBPuvT5d-I/edit?usp=sharing

Currently blocked by the new protocol being in draft status and migration steps not yet public.

**Comment by Azier777** [comment]
What does T13-blocked mean

**Comment by lazaridiscom** [comment]
> What does T13-blocked mean

Just a label that this issue is blocked ("Currently blocked by the new protocol being in draft status and migration steps not yet public.").

**Comment by danfinlay** [comment]
Please resist the urge to turn every issue into a critique of our internal organization methodology.

Many of our tags of category prefixes. You can ignore them.

I believe a usable v3 manifest build is coming soon, so we will be able to pursue this soon. @kumavis was looking at this recently.

I will also share [An Open Letter to the Chrome Team About Their Proposed v3 Changes to the WebExtension Format](https://medium.com/metamask/an-open-letter-to-the-chromium-team-about-their-proposed-changes-to-the-webextension-standard-8117f2acf32).

**Comment by lazaridiscom** [comment]
> Please resist the urge to turn every issue into a critique of our internal organization methodology.

You're internal organization methodology is fine (from what I can see), as is the user-facing one in this issue-tracker.

**Comment by robertdavid010** [comment]
Has there been any update on this? I've been looking into this issue more in depth recently, it seems like the discussion didn't continue? Or where is the current discussion at?

**Comment by danfinlay** [comment]
The migration guide has been published, we are no longer blocked on beginning this conversion:
https://developer.chrome.com/docs/extensions/mv3/intro/mv3-migration/

**Comment by danfinlay** [comment]
[An article on Firefox support for manifest v3](https://blog.mozilla.org/addons/2021/05/27/manifest-v3-update/)

**Comment by santdeleon** [comment]
How is MetaMask planning on storing the unencrypted mnemonic once persistence is removed from the background page in v3 extensions?

**Comment by Edison4mobile** [comment]
Hi, metamask team. you guys are planing to migrate manifest v3 soon?

**Comment by jpuri** [comment]
Here is internal document with implementation details: https://www.notion.so/6f394364046d426690471a977aec1e6f?v=6bb3deb1e14243a5b34b390ea3292dff&p=19063a3ee9af46c48f18bf8e8f7aa546

**Comment by sondreb** [comment]
@jpuri - Could that document be made public available?

Any additional information on the upgrade? It is now February 2022, that means it's less than one year until Chrome will stop working with Manifest V2 extensions.

**Comment by brad-decker** [comment]
Going to close this issue, we have prepared and are working on rolling out MV3 support for chrome. A new issue will be made to track towards Firefox support.
